### PR TITLE
util/bytes: make unprefixedHexToBytes specific

### PR DESCRIPTION
**Pull request by jochem-brouwer**
This PR adds an extra guard to the proxy `hexToBytes` from `ethereum-cryptography`. That method supports both prefixed and unprefixed hex strings. The new (deprecated) `unprefixedHexToBytes` now explicitly only accepts unprefixed hex strings.

(Tests seem to pass locally so ready for review)

Follow up of #2845

### CODE REVIEW

File: packages/evm/src/precompiles/06-ecadd.ts
Comment (jochem-brouwer) [inline packages/evm/src/precompiles/06-ecadd.ts:34]: Note: (for all `ec*` files), it seems that if there is a return value, the string is not prefixed, but if there is no return value (error?) then 0x is returned. (So we need to do this check)

**Comment by jochem-brouwer** [comment]
Just verified (see prev. commit): all tests complete on if we merge https://github.com/ethereumjs/rustbn-wasm/pull/5. This PR is on hold until it is merged and released.

**Comment by jochem-brouwer** [comment]
(this PR will need an update to that new pkg version in order to run CI)

File: packages/util/src/bytes.ts
Comment (holgerd77) [inline packages/util/src/bytes.ts:24]: We can‘t do this check here since an unprefixed hex string can very well start with 0x. 😅 🤯

That‘s one of the reasons why we take prefixed as preferred: one can have a positive check that the format is correct.

**Review by holgerd77 — CHANGES_REQUESTED** [review]

File: packages/util/src/bytes.ts
Comment (holgerd77) [reply packages/util/src/bytes.ts:24]: Actually that also means that you cannot use the method from ethereum-cryptography, now that I am thinking about it, right? So what you can do is always to add 0x and then use our own prefixed method I would think.

Comment (jochem-brouwer) [reply packages/util/src/bytes.ts:24]: Sorry, I thought we decided we wanted to explicitly check here if it was unprefixed? But, I can change, and then call into our own method (so we do not import `hexToBytes` from `ethereum-cryptography` anymore)

Comment (holgerd77) [reply packages/util/src/bytes.ts:24]: Ah wait, maybe let's give this another round. Or would the Ox-in-string case only count for the utf8ToBytes version? 🤔

Comment (acolytec3) [reply packages/util/src/bytes.ts:24]: >We can‘t do this check here since an unprefixed hex string can very well start with 0x

How is this possible, or have I misunderstood something along the way?  Hex doesn't include `x`, right, so unprefixed would explicitly mean no `0x`, right?

Comment (holgerd77) [reply packages/util/src/bytes.ts:24]: Sorry, I think I am actually off here all the way down.

Have to some extend mixed this up with arbitrary (UTF-8) strings (fee very free to challenge me more on these kind of things, I stated this explicitly in the whole reasoning for the prefixed-hex work).

Anyhow: I think the outcome is (gladly enough, puh) still desired to have this unified since this reduces the risk of errrors when working in an inconsistent way with prefixed and unprefixed hex strings.

Ok, will update the branch here and merge.

**Comment by holgerd77** [comment]
Rebased this via UI

**Review by holgerd77 — APPROVED** [review]
LGTM

**Pull request merged**
